mora/service/orgunit.py

```diff
@@ -0,0 +1,647 @@
+
+    return r
+
+
+@blueprint.route('/<any(o,ou):type>/<uuid:parentid>/children')
+@util.restrictargs('at')
+def get_children(type, parentid):
+    '''Obtain the list of nested units within an organisation or an
+    organisational unit.
+
+    .. :quickref: Unit; Children
+
+    :param type: 'o' if the parent is an organistion, and 'ou' if it's a unit.
+    :param uuid parentid: The UUID of the parent.
+
+    :queryparam date at: Current time in ISO-8601 format.
+
+    :<jsonarr string name: Human-readable name of the unit.
+    :<jsonarr string user_key: Short, unique key identifying the unit.
+    :<jsonarr uuid uuid: Machine-friendly UUID of the unit.
+    :<jsonarr int child_count: Number of org. units nested immediately beneath
+                               the organisation.
+
+    :status 200: Whenever the organisation or unit exists and is readable.
+    :status 404: When no such organisation or unit exists, or the
+                 parent was of the wrong type.
+
+    **Example Response**:
+
+    .. sourcecode:: json
+
+      [
+        {
+          "name": "Humanistisk fakultet",
+          "user_key": "hum",
+          "uuid": "9d07123e-47ac-4a9a-88c8-da82e3a4bc9e",
+          "child_count": 2
+        },
+        {
+          "name": "Samfundsvidenskabelige fakultet",
+          "user_key": "samf",
+          "uuid": "b688513d-11f7-4efc-b679-ab082a2055d0",
+          "child_count": 0
+        }
+      ]
+
+    '''
+    c = common.get_connector()
+
+    if type == 'o':
+        scope = c.organisation
+    else:
+        assert type == 'ou'
+        scope = c.organisationenhed
+
+    obj = scope.get(parentid)
+
+    if not obj or not obj.get('attributter'):
+        raise werkzeug.exceptions.NotFound
+
+    children = [
+        get_one_orgunit(c, childid, child)
+        for childid, child in
+        c.organisationenhed.get_all(overordnet=parentid,
+                                    gyldighed='Aktiv')
+    ]
+
+    children.sort(key=operator.itemgetter('name'))
+
+    return flask.jsonify(children)
+
+
+@blueprint.route('/ou/<uuid:unitid>/')
+@util.restrictargs('at', 'validity')
+def get_orgunit(unitid):
+    '''Retrieve an organisational unit.
+
+    .. :quickref: Unit; Get
+
+    :queryparam date at: Current time in ISO-8601 format.
+    :queryparam string validity: Only show *past*, *present* or
+        *future* values -- which the default being to show *present*
+        values.
+
+    :<jsonarr string name: Human-readable name.
+    :<jsonarr string uuid: Machine-friendly UUID.
+    :<jsonarr string user_key: Short, unique key identifying the unit.
+    :<jsonarr object org: The organisation that this unit belongs to, as
+        yielded by :http:get:`/service/o/`.
+    :<jsonarr object validity: The validity of this entry.
+
+    :status 404: If the organisational unit isn't found.
+    :status 200: Otherwise.
+
+    **Example Response**:
+
+    .. sourcecode:: json
+
+      [
+        {
```

Comment:
We should also return `org_unit_type` and `parent`

Reply:
Good point! I'll add that in a followup so that @n-singh has something to work with :)